Zendesk Mobile: show notification alert on navigation bar Help

ScoutHarris opened the pull request:
Fixes #9338 

This PR does two things:
1) When a ZD PN is received, it adds an orange indicator on the navigation bar Help button.
2) All the relevant VCs observe ZD `NSNotification`s for PNs received and PNs cleared. This allows all the notification indicators to be in sync, ultimately dependent on the `ZendeskUtils:showSupportNotificationIndicator` value. 

NOTE: This does not fix the issue with the Notifications tab showing an indicator when ZD PNs are received. That's still to come.

**To test:**
**1) Nav bar Help button indicator**
- Go to 'Support' and submit a ticket.
- In Zendesk, reply to your ticket (or ask me to).
- In the app, go to site creation (either WP or self-hosted).
  - Verify the indicator is shown.
- Click 'Help'
  - Verify the indicator is shown on 'My Tickets'.
- Select 'My Tickets', then close it.
  - Verify the indicator is no longer shown.
- Close to return to site creation.
  - Verify the 'Help' indicator is no longer shown.

To note, If you are in site creation when a new PN is received, the Zendesk PN alert will appear. Viewing a ticket from the ZD PN alert (no matter where it is displayed) will also clear the indicators.

**2) Indicator syncing**
Instead of listing all the possible scenarios, I'll just describe the behavior.

These indicators should all be in the same state:
- Me tab icon
- Me tab > Help & Support
- Help & Support > My Tickets
- Navigation bar Help

When a new PN is received, all these indicators should be present.
Once 'My Tickets' is selected, all these indicators should be cleared.

**Help button indicator:**
<img width="92" alt="help_indicator" src="https://user-images.githubusercontent.com/1816888/40025956-a13599cc-5790-11e8-8325-13254bdd0ac7.png">

jleandroperez commented on WordPressAuthenticator/WordPressAuthenticator/NUX/WPHelpIndicatorView.swift:
```diff
@@ -0,0 +1,36 @@
+open class WPHelpIndicatorView: UIView {
+
+    struct Constants {
+        static let defaultInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
```
Tip tip! this is equivalent to `UIEdgeInsets.zero`.

There's also `CGRect.zero` and `CGPoint.zero`. Neat!!

ScoutHarris replied:
Ah, good catch. It shall be changed!

ScoutHarris commented:
@aerych - I updated the indicator placement per @SylvesterWilmott 's feedback. The center is now (-5, 3) from the Help button top right corner. I've updated the screenshot in the description as well.

Do you want to take another look? If you want real PNs we can do that. If you just want to see the indicator you can set `helpIndicator.isHidden = false` in `NUXViewControllerBase:refreshSupportNotificationIndicator`.

aerych commented:
If @SylvesterWilmott is happy with it then its 👍 by me. :)
